### PR TITLE
Fixes for recent merges

### PR DESCRIPTION
**Pull request by thinkyhead**
Fix bug introduced by #7276, apply coding standards, correct indentation, remove trailing whitespace, correct spelling, use shorthand macros where possible, strip dead code, etc.

[Concise diff](https://github.com/MarlinFirmware/Marlin/pull/7281/files?w=1)

### CODE REVIEW

File: Marlin/Marlin_main.cpp [hunk 1]
```diff
@@ -4617,11 +4617,11 @@ void home_all_axes() { gcode_G28(true); }
 
         #if ENABLED(AUTO_BED_LEVELING_LINEAR)
 
-//        mean += measured_z;                                  // I believe this is unused code?
-//        eqnBVector[abl_probe_index] = measured_z;            // I believe this is unused code?
-//        eqnAMatrix[abl_probe_index + 0 * abl2] = xProbe;     // I believe this is unused code?
-//        eqnAMatrix[abl_probe_index + 1 * abl2] = yProbe;     // I believe this is unused code?
-//        eqnAMatrix[abl_probe_index + 2 * abl2] = 1;          // I believe this is unused code?
```
Comment (thinkyhead) [inline Marlin/Marlin_main.cpp:4624]: very much used code

File: Marlin/Marlin_main.cpp [hunk 2]
```diff
@@ -4797,9 +4797,6 @@ void home_all_axes() { gcode_G28(true); }
 
               incremental_LSF(&lsf_results, xProbe, yProbe, measured_z);
 
-          #if ENABLED(AUTO_BED_LEVELING_LINEAR)
-            indexIntoAB[xCount][yCount] = abl_probe_index;
-          #endif
```
Comment (thinkyhead) [inline Marlin/Marlin_main.cpp:4802]: whoops

File: Marlin/Marlin_main.cpp (hunk 1 above)
Comment (Roxy-3D) [reply Marlin/Marlin_main.cpp:4624]: The bigger context is show here:
```cpp
        #if ENABLED(AUTO_BED_LEVELING_LINEAR)
//        mean += measured_z;                                  // I believe this is unused code?
//        eqnBVector[abl_probe_index] = measured_z;            // I believe this is unused code?
//        eqnAMatrix[abl_probe_index + 0 * abl2] = xProbe;     // I believe this is unused code?
//        eqnAMatrix[abl_probe_index + 1 * abl2] = yProbe;     // I believe this is unused code?
//        eqnAMatrix[abl_probe_index + 2 * abl2] = 1;          // I believe this is unused code?

        #elif ENABLED(AUTO_BED_LEVELING_BILINEAR)
```

I put a SERIAL_ECHO() in that block of code and it never happened (with AUTO_BED_LEVELING_LINEAR turned on).    I would encourage you to try the same thing.

Comment (Roxy-3D) [reply Marlin/Marlin_main.cpp:4624]: Oh!   I see this code block happens with a 'Q' option.    So in that case, the fix is a little bit more involved.    We need to also call this for each point sampled in this block.
```cpp
             incremental_LSF(&lsf_results, xProbe, yProbe, measured_z);
```

**Pull request merged**